code/datums/helper_datums/teleport.dm

```diff
@@ -113,9 +113,76 @@
 		var/mob/living/L = teleatom
 		if(L.buckled)
 			C = L.buckled
+
 	if(force_teleport)
 		teleatom.forceMove(destturf)
 		playSpecials(destturf,effectout,soundout)
+		var/atom/impediment
+
+		if(destturf.density)
+			impediment = destturf
+
+		else
+			for(var/atom/A in destturf)
+				if(A.density)
+					impediment = A
+					break
+
+		if(impediment)
+			var/turf/newdest
+			var/boominess = 0
+			for(var/turf/T in range(1, destturf))
+				if(!T.density)
+					for(var/atom/A in T)
+						var/blocked = 0
+						if(A.density && A.opacity)
+							blocked = 1
+							break
+						if(!blocked)
+							newdest = T
+							break
+
+			if(istype(teleatom, /obj))
+				var/obj/O = teleatom
+				if(newdest)
+					O.forceMove(newdest)
+					O.ex_act(3)
+				else
+					O.crush_act()
+				boominess += O.w_class
+				if(O.density)
+					boominess += 5
+				if(O.opacity)
+					boominess += 10
+
+			if(istype(teleatom, /mob/living))
+				var/mob/living/L = teleatom
+				boominess += L.mob_size/4
+				if(istype(L, /mob/living/carbon/human))
+					var/mob/living/carbon/human/H = L
+					if(newdest)
+						var/obj/item/organ/external/E = pick(H.organs)
+						to_chat(H, "<span class='danger'>You partially phase into \the [impediment], causing your \the [E] to bloodily dematerialize!</span>")
+						E.droplimb(0,DROPLIMB_BLUNT)
+						H.forceMove(newdest)
+					else
+						to_chat(H, "<span class='danger'>Your life flashes before your eyes as you phase into \the [impediment] before the universe suddenly and violently corrects itself!</span>")
+						H.gib()
+				else
+					if(newdest)
+						to_chat(L, "<span class='danger'>You partially phase into \the [impediment], causing a chunk of you to bloodily dematerialize!</span>")
+						L.adjustBruteLoss(40)
+						L.forceMove(newdest)
+					else
+						to_chat(L, "<span class='danger'>Your life flashes before your eyes as you phase into \the [impediment] before the universe suddenly and violently corrects itself!</span>")
+						L.gib()
+
+			if(!newdest)
+				boominess += 10
+
+			destturf.visible_message("<span class ='danger'>There is a sizable emission of energy as \the [teleatom] phases into \the [impediment]!</span>")
+			explosion(destturf, ((boominess > 10) ? 1 : 0), ((boominess > 5) ? (boominess/5) : 0), boominess/5, boominess/2)

```

Comment:
I am entirely against the idea of any sort of explosion, unless we teleport volatile things into dense object. Even on that case explosion is very low chance.

According to how teleportation works - you disassemble atoms in one place and assemble into another place, the collision of atoms results into what Github calls merge conflict. Result of the conflict makes two atoms merge into one atom creating a hybrid of both object. The explosion should never occur unless two object contained atoms that in combination create combustion and tiny explosion(by tiny we are talking atom size explosion).

Reply:
I disagree.

Reply:
What do you mean desire? First we do not have lore about teleportation, because I checked the wiki page for it and nothing mentions how it works. Everything is always explained as bluespace. If the time is involved into teleportation then I can agree that explosions might occur. But the only way it can at that point is if at certain point of time this atom had something that would create combustion with current atom. If we take into account the alternate dimensions that could justify it - because in some other universe on place of that wall can be fuel tank or something else. But again given that this theory leads us that alternate dimensions are infinite - that means that chances of having explosion are close to 0.

To summarize: with your justification of why it happens you need to make explosion a low probability thing.

Reply:
No.

Reply:
What do you mean no?

Reply:
I mean that I'm not adding in a "probability" of explosions, I do not find your justifications convincing, and further I don't really see a need to justify the explosive nature of energy emission any further than I already have. The brass tacks can be divvied up by someone who cares about that sort of thing.